docs/migrating.md

```diff
@@ -100,6 +100,9 @@ however there are some known differences to take into account, see:
 in the LLD documentation.
 * The [LLD and GNU linker incompatibilities](https://maskray.me/blog/2020-12-19-lld-and-gnu-linker-incompatibilities)
 blog post.
+* When selecting the printf/scanf library variant with `--defsym from=to`, lld includes not only the specified
```

Comment:
add a reference to https://github.com/picolibc/picolibc/blob/main/doc/printf.md after the defsym to be clear when/how it is used.